bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/obsComp/TargetObsComp.java

```diff
@@ -90,14 +91,18 @@ public String getTitle() {
             // assume user did not edit title manually
             TargetEnvironment env = getTargetEnvironment();
             SPTarget tp = env.getBase();
-            if (tp != null) { // base pos should always be defined, so tp should never be null
-                String name = tp.getTarget().getName();
-                if (name == null || name.length() == 0) {
-                    // Use the base position
-                    return tp.getTarget().getPosition();
+            if (tp != null) {
+                ITarget t = tp.getTarget();
+                String name = t.getName();
+                if (name == null || name.trim().length() == 0) {
+                    name = "<Untitled>";
+                }
+                switch (t.getTag()) {
+                    case JPL_MINOR_BODY:   return "Comet: " + name;
+                    case MPC_MINOR_PLANET: return "Minor Planet: " + name;
+                    case NAMED:            return "Solar System Object: " + name;
+                    case SIDEREAL:         return "Sidereal: " + name;
```

Comment:
The `position` method is vestigial and made no sense at all (see the old impl on `ConicTarget` below). I talked to Andy and we decided to try this and see what people say. If we had s distinct icon for each target type we could get rid of the "type" prefix. I opened an issue for this.